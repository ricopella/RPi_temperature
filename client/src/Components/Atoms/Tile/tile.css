.tile {
    background-color: #A9A9A9;
    margin: 20px 25vw;
    padding: 20px 20px;
    height: 700px;
    width: 600px;
}