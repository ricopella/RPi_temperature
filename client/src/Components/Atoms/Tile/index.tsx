import * as React from 'react';
import ClassNames from '../../../utils/classNames';

import './tile.css';

interface TileProps {
  children?: React.ReactNode | React.ReactNodeArray;
  className?: string;
  style?: React.CSSProperties;
}

const Tile: React.SFC<TileProps> = ({ children, className }) => {
  return (
    <div className={ClassNames({ tile: true, className })}>
      {children && children}
    </div>
  );
};

export default Tile;
