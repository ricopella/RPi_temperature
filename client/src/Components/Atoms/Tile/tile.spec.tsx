import { mount } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Tile from '.';

describe('<Tile />', () => {
  it('should match internal tree', () => {
    const internalTree = renderer.create(<Tile />).toJSON();

    expect(internalTree).toMatchSnapshot();
  });

  it('should have one child', () => {
    const component = mount(
      <Tile>
        <div className="child" />
      </Tile>,
    );

    const actual = component.find('.child').length;
    const expected = 1;

    expect(actual).toEqual(expected);
  });
});
