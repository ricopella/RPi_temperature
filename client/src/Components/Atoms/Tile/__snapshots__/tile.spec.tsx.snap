// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tile /> should match internal tree 1`] = `
<div
  className="tile"
/>
`;
